### PR TITLE
Fi menu overhaul

### PR DESCRIPTION
**Pull request by Muzugalium**

### CODE REVIEW

**Comment by Muzugalium** [comment]
Found a bug with the dungeon status where maps are always shown as unobtained, will try and fix tomorrow

File: asm/custom-functions/src/lib.rs [hunk 1]
```diff
@@ -219,6 +238,77 @@ fn handle_bk_map_dungeonflag(item: c_ushort) {
...
+fn rando_text_command_handler(_event_flow_mgr: *mut c_void, p_flow_element: *const FlowElement) {
+    let flow_element = unsafe { &*p_flow_element };
+    match flow_element.param3 {
+        71 => {
+            let dungeon_index = flow_element.param1;
+            let completion_storyflag = flow_element.param2;
+            let mut key_count = dungeon_global_key_count(dungeon_index);
+            if dungeon_index == 14 {
+                unsafe { key_count = getKeyPieceCount();}
+            }
```
Comment (lepelog) [inline asm/custom-functions/src/lib.rs:257]:
```suggestion
            let key_count = if dungeon_index == 14 /* ET */ {
                unsafe { getKeyPieceCount() }
            } else {
                dungeon_global_key_count(dungeon_index)
            };
```

File: gamepatches.py [hunk 2]
```diff
@@ -1648,42 +1651,110 @@ def add_required_dungeon_patches(self):
...
             combined = ""
 
             for part in required_dungeons_text.split(" "):
-                if len(cur_line + part) > 27:  # limit of one line
+                # limit of one line (notice board = 27, textbox = 36)
+                if len(cur_line + part) > 36:
                     combined += cur_line + "\n"
                     cur_line = part + " "
                 else:
                     cur_line += part + " "
```
Comment (lepelog) [inline gamepatches.py:1681]: Use `break_lines` from `textbox_utils`(it didn't exist when this code was written, but would be a good cleanup)

**Review by lepelog — CHANGES_REQUESTED** [review]

File: gamepatches.py (hunk 2 above)
Comment (Muzugalium) [reply gamepatches.py]: Done ^^

File: asm/custom-functions/src/lib.rs (hunk 1 above)
Comment (Muzugalium) [reply asm/custom-functions/src/lib.rs]: Also done ^^

**Comment by CovenEsme** [comment]
Tested in race today - no issues found.

**Pull request merged**